src/test/java/api/requests/RequestScheduledNoticesTests.java

```diff
@@ -0,0 +1,342 @@
+
+    useLoanPolicyAsFallback(
+      loanPoliciesFixture.canCirculateRolling().getId(),
+      requestPoliciesFixture.pageRequestPolicy().getId(),
+      noticePoliciesFixture.create(noticePolicyBuilder).getId());
+
+    requestsFixture.place(new RequestBuilder().page()
+      .forItem(item)
+      .withRequesterId(requester.getId())
+      .withRequestDate(DateTime.now())
+      .withStatus(OPEN_NOT_YET_FILLED)
+      .withPickupServicePoint(pickupServicePoint)
+      .withRequestExpiration(null));
+
+    TimeUnit.SECONDS.sleep(1);
+
+    List<JsonObject> scheduledNotices = scheduledNoticesClient.getAll();
+    assertThat(scheduledNotices, hasSize(0));
+  }
+
+  @Test
+  public void requestExpirationBeforeNoticeShouldBeScheduledOnRequestCreate()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    JsonObject requestNotice = new NoticeConfigurationBuilder()
+      .withTemplateId(templateId)
+      .withRequestExpirationEvent()
+      .withBeforeTiming(Period.days(3))
+      .sendInRealTime(true)
+      .create();
+
+    NoticePolicyBuilder noticePolicyBuilder = new NoticePolicyBuilder()
+      .withName("request policy")
+      .withRequestNotices(Collections.singletonList(requestNotice));
+
+    useLoanPolicyAsFallback(
+      loanPoliciesFixture.canCirculateRolling().getId(),
+      requestPoliciesFixture.pageRequestPolicy().getId(),
+      noticePoliciesFixture.create(noticePolicyBuilder).getId());
+
+    LocalDate requestExpiration = LocalDate.now().plusMonths(3);
+    IndividualResource request = requestsFixture.place(new RequestBuilder().page()
+      .forItem(item)
+      .withRequesterId(requester.getId())
+      .withRequestDate(DateTime.now())
+      .withStatus(OPEN_NOT_YET_FILLED)
+      .withPickupServicePoint(pickupServicePoint)
+      .withRequestExpiration(requestExpiration));
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(scheduledNoticesClient::getAll, hasSize(1));
+
+    List<JsonObject> scheduledNotices = scheduledNoticesClient.getAll();
+    assertThat(scheduledNotices.size(), is(1));
+
+    JsonObject scheduledNotice = scheduledNotices.get(0);
+    JsonObject noticeConfig = scheduledNotice.getJsonObject("noticeConfig");
+
+    assertThat(scheduledNotice.getString("requestId"), is(request.getId().toString()));
+    assertThat(scheduledNotice.getString("triggeringEvent"), is("Request Expiration"));
+    assertThat(scheduledNotice.getString("nextRunTime"),
+      isEquivalentTo(requestExpiration.toDateTimeAtStartOfDay().minusDays(3)));
+    assertThat(noticeConfig.getString("timing"), is("Before"));
+    assertThat(noticeConfig.getString("templateId"), is(templateId.toString()));
+    assertThat(noticeConfig.getString("format"), is("Email"));
+    assertThat(noticeConfig.getBoolean("sendInRealTime"), is(true));
+  }
+
+  @Test
+  public void requestExpirationUponAtNoticeShouldBeRescheduledOnRequestUpdate()
```

Comment:
What happens in the situation where the expiration date is removed during an update? Is there a test for that?

Reply:
In this case scheduled notice should be removed. Added a test for that.